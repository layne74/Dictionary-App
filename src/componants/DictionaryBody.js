import React from 'react'

export default function DictionaryBody( props ) {
  // Props deconstruction
  const {word, realWord, definition, example, clickFunc, changeFunc, tempWord} = props;

  // Based off the props, either a result, prompt or error will be returned
  function toReturn(){
      // If nothing is entered, the promp is returned
      if (definition === null && realWord === true) {
          return (
            <div>
              <h3>Search for a word!</h3>
            </div>
          )
      // If the word has been found, it is returned
      } else if (realWord === true) {
          return (
            <div>
              <h3>Definition:</h3>
              <p>{definition}</p>
              <h3>Example:</h3>
              <p>{example}</p>
            </div>
          )
      // If the word has not been found a error is displayed
      } else {
          // Creates a link so the user can quickly google what they are looking for
          const googleLink = "https://www.google.co.za/search?q=" + tempWord + " definition"
          return (
            <div>
              <h3>Hmm... I cant seem to find that word. Try <a href={googleLink} target="blank">google</a> it.</h3>
            </div>
          )
      }
  }

  return (
      <div>
          <form onSubmit={clickFunc}>
                <input onChange={changeFunc} value={word} placeholder="Enter a word" required></input><br></br>
                <button type="submit">Search</button>
          </form>
          {toReturn()}
      </div>
  )
    
}
