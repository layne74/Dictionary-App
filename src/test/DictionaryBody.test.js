import React from 'react';
import renderer from 'react-test-renderer';
import DictionaryBody from '../componants/DictionaryBody';

test('renders correctly', () => {
    const tree = renderer
    .create(<DictionaryBody 
        word="gold" 
        realWord={true} 
        definition="Some Definition" 
        example="Some Example"/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});