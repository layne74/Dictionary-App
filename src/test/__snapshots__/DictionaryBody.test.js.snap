// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form>
    <input
      placeholder="Enter a word"
      required={true}
      value="gold"
    />
    <br />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
  <div>
    <h3>
      Definition:
    </h3>
    <p>
      Some Definition
    </p>
    <h3>
      Example:
    </h3>
    <p>
      Some Example
    </p>
  </div>
</div>
`;
